nltk/stem/paicehusk.py

```diff
@@ -0,0 +1,274 @@
+    jrev1t.   { -verj > -vert }
+    jsim2t.   { -misj > -mit }
+    jn1d.     { -nj > -nd  }
+    j1s.      { -j > -s    }
+    lbaifi6.  { -ifiabl > - }
+    lbai4y.   { -iabl > -y }
+    lba3>     { -abl > -   }
+    lbi3.     { -ibl > -   }
+    lib2l>    { -bil > -bl }
+    lc1.      { -cl > c    }
+    lufi4y.   { -iful > -y }
+    luf3>     { -ful > -   }
+    lu2.      { -ul > -    }
+    lai3>     { -ial > -   }
+    lau3>     { -ual > -   }
+    la2>      { -al > -    }
+    ll1.      { -ll > -l   }
+    mui3.     { -ium > -   }
+    mu*2.     { -um > -   if intact }
+    msi3>     { -ism > -   }
+    mm1.      { -mm > -m   }
+    nois4j>   { -sion > -j }
+    noix4ct.  { -xion > -ct }
+    noi3>     { -ion > -   }
+    nai3>     { -ian > -   }
+    na2>      { -an > -    }
+    nee0.     { protect  -een }
+    ne2>      { -en > -    }
+    nn1.      { -nn > -n   }
+    pihs4>    { -ship > -  }
+    pp1.      { -pp > -p   }
+    re2>      { -er > -    }
+    rae0.     { protect  -ear }
+    ra2.      { -ar > -    }
+    ro2>      { -or > -    }
+    ru2>      { -ur > -    }
+    rr1.      { -rr > -r   }
+    rt1>      { -tr > -t   }
+    rei3y>    { -ier > -y  }
+    sei3y>    { -ies > -y  }
+    sis2.     { -sis > -s  }
+    si2>      { -is > -    }
+    ssen4>    { -ness > -  }
+    ss0.      { protect  -ss }
+    suo3>     { -ous > -   }
+    su*2.     { -us > -   if intact }
+    s*1>      { -s > -    if intact }
+    s0.       { -s > -s    }
+    tacilp4y. { -plicat > -ply }
+    ta2>      { -at > -    }
+    tnem4>    { -ment > -  }
+    tne3>     { -ent > -   }
+    tna3>     { -ant > -   }
+    tpir2b.   { -ript > -rib }
+    tpro2b.   { -orpt > -orb }
+    tcud1.    { -duct > -duc }
+    tpmus2.   { -sumpt > -sum }
+    tpec2iv.  { -cept > -ceiv }
+    tulo2v.   { -olut > -olv }
+    tsis0.    { protect  -sist }
+    tsi3>     { -ist > -   }
+    tt1.      { -tt > -t   }
+    uqi3.     { -iqu > -   }
+    ugo1.     { -ogu > -og }
+    vis3j>    { -siv > -j  }
+    vie0.     { protect  -eiv }
+    vi2>      { -iv > -    }
+    ylb1>     { -bly > -bl }
+    yli3y>    { -ily > -y  }
+    ylp0.     { protect  -ply }
+    yl2>      { -ly > -    }
+    ygo1.     { -ogy > -og }
+    yhp1.     { -phy > -ph }
+    ymo1.     { -omy > -om }
+    ypo1.     { -opy > -op }
+    yti3>     { -ity > -   }
+    yte3>     { -ety > -   }
+    ytl2.     { -lty > -l  }
+    yrtsi5.   { -istry > - }
+    yra3>     { -ary > -   }
+    yro3>     { -ory > -   }
+    yfi3.     { -ify > -   }
+    ycn2t>    { -ncy > -nt }
+    yca3>     { -acy > -   }
+    zi2>      { -iz > -    }
+    zy1s.     { -yz > -ys  }
+    """
+
+    def __init__(self, ruletable=None):
+        """
+        :param ruletable: a string containing the rule data, separated
+            by newlines.
+        """
+        self.rules = defaultdict(list)
+        if ruletable:
+            self.read_rules(ruletable)
+        else:
+            self.read_rules(self.defaultrules)
+
```

Comment:
It'll be good to add the ruletable variable to the stemmer object that way, it can be modified later on (if needed), e.g. 

```
self._ruletable = ruletable if ruletable else self.defaultrules
self.read_rules(self._ruletable)
```